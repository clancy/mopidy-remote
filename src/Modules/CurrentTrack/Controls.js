import React, { Component, PropTypes } from 'react';
import {
  View,
  StyleSheet,
  TouchableOpacity
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons'

const playIconName = paused => paused ? 'play-arrow' : 'pause';
const iconStyle = enabled => {
  return {
      marginLeft: 10,
      marginRight: 10,
      'color' : enabled ? 'green' : 'white'
  };
};

export default Controls = React.createClass({
  propTypes: {
    paused: PropTypes.bool.isRequired,
    shuffle: PropTypes.bool.isRequired,
    repeat: PropTypes.bool.isRequired,
    playPause: PropTypes.func.isRequired,
    previous: PropTypes.func.isRequired,
    next: PropTypes.func.isRequired
  },
  render() {
    return (
      <View style={styles.container}>
        <TouchableOpacity>
          <Icon name='shuffle' size={25} style={iconStyle(this.props.shuffle)} />
        </TouchableOpacity>
        <TouchableOpacity
          onPress={this.props.previous} >
          <Icon name='skip-previous' size={35} style={iconStyle()} />
        </TouchableOpacity>
        <View style={styles.playIconContainer}>
          <TouchableOpacity
            onPress={this.props.playPause} >
            <Icon name={playIconName(this.props.paused)} size={50} style={styles.playIcon} />
          </TouchableOpacity>
        </View>
        <TouchableOpacity
          onPress={this.props.next}>
          <Icon name='skip-next' size={35} style={iconStyle()}/>
        </TouchableOpacity>
        <TouchableOpacity>
          <Icon name='repeat' size={25} style={iconStyle(this.props.repeat)}/>
        </TouchableOpacity>
      </View>
    );
  }
});

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    backgroundColor: 'black',
    justifyContent: 'center',
    alignItems: 'center'
  },
  playIcon: {
    color: 'white',
    backgroundColor: 'black',
    backgroundColor: 'transparent'
  },
  playIconContainer: {
    borderWidth: StyleSheet.hairlineWidth,
    borderRadius: 35,
    width: 70,
    height: 70,
    borderColor: 'white',
    justifyContent: 'center',
    alignItems: 'center'
  }
});
