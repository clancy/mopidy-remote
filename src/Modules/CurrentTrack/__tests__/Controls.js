import 'react-native';
import React from 'react';
import Controls from '../Controls';
import { shallow, mount, render } from 'enzyme';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('Controls tests', () => {
  test('renders disabled icons correctly', () => {
    const tree = renderer.create(
      <Controls
        previous={() => {}}
        next={() => {}}
        shuffle={() => {}}
        repeat={() => {}}
        playPause={() => {}}
        playing={false}
        shuffleEnabled={false}
        repeatEnabled={false} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders enabled icons correctly', () => {
    const tree = renderer.create(
      <Controls
        previous={() => {}}
        next={() => {}}
        shuffle={() => {}}
        repeat={() => {}}
        playPause={() => {}}
        playing={true}
        shuffleEnabled={true}
        repeatEnabled={true} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('connects shuffle control correctly', () => {
    var controlMock = jest.fn();
    const controls = shallow(
      <Controls
        previous={() => {controlMock('previous')}}
        next={() => {controlMock('next')}}
        shuffle={() => {controlMock('shuffle')}}
        repeat={() => {controlMock('repeat')}}
        playPause={() => {controlMock('playPause')}}
        playing={true}
        shuffleEnabled={true}
        repeatEnabled={true} />
    );

    let shuffleButton = controls.findWhere(s => s.name() == 'TouchableOpacity' && s.children().at(0).prop('name') == 'shuffle').at(0);
    shuffleButton.simulate('press');
    expect(controlMock.mock.calls.length).toBe(1);
    expect(controlMock.mock.calls[0][0]).toBe('shuffle');
  });

  test('connects previous control correctly', () => {
    var controlMock = jest.fn();
    const controls = shallow(
      <Controls
        previous={() => {controlMock('previous')}}
        next={() => {controlMock('next')}}
        shuffle={() => {controlMock('shuffle')}}
        repeat={() => {controlMock('repeat')}}
        playPause={() => {controlMock('playPause')}}
        playing={true}
        shuffleEnabled={true}
        repeatEnabled={true} />
    );

    let previousButton = controls.findWhere(s => s.name() == 'TouchableOpacity' && s.children().at(0).prop('name') == 'skip-previous').at(0);
    previousButton.simulate('press');
    expect(controlMock.mock.calls.length).toBe(1);
    expect(controlMock.mock.calls[0][0]).toBe('previous');
  });

  test('connects next control correctly', () => {
    var controlMock = jest.fn();
    const controls = shallow(
      <Controls
        previous={() => {controlMock('previous')}}
        next={() => {controlMock('next')}}
        shuffle={() => {controlMock('shuffle')}}
        repeat={() => {controlMock('repeat')}}
        playPause={() => {controlMock('playPause')}}
        playing={true}
        shuffleEnabled={true}
        repeatEnabled={true} />
    );

    let nextButton = controls.findWhere(s => s.name() == 'TouchableOpacity' && s.children().at(0).prop('name') == 'skip-next').at(0);
    nextButton.simulate('press');
    expect(controlMock.mock.calls.length).toBe(1);
    expect(controlMock.mock.calls[0][0]).toBe('next');
  });

  test('connects repeat control correctly', () => {
    var controlMock = jest.fn();
    const controls = shallow(
      <Controls
        previous={() => {controlMock('previous')}}
        next={() => {controlMock('next')}}
        shuffle={() => {controlMock('shuffle')}}
        repeat={() => {controlMock('repeat')}}
        playPause={() => {controlMock('playPause')}}
        playing={true}
        shuffleEnabled={true}
        repeatEnabled={true} />
    );

    let repeatButton = controls.findWhere(s => s.name() == 'TouchableOpacity' && s.children().at(0).prop('name') == 'repeat').at(0);
    repeatButton.simulate('press');
    expect(controlMock.mock.calls.length).toBe(1);
    expect(controlMock.mock.calls[0][0]).toBe('repeat');
  });

  test('connects playPause control correctly', () => {
    var controlMock = jest.fn();
    const controls = shallow(
      <Controls
        previous={() => {controlMock('previous')}}
        next={() => {controlMock('next')}}
        shuffle={() => {controlMock('shuffle')}}
        repeat={() => {controlMock('repeat')}}
        playPause={() => {controlMock('playPause')}}
        playing={true}
        shuffleEnabled={true}
        repeatEnabled={true} />
    );

    let playPauseButton = controls.findWhere(s => s.name() == 'TouchableOpacity' && s.children().at(0).prop('name') == 'pause').at(0);
    playPauseButton.simulate('press');
    expect(controlMock.mock.calls.length).toBe(1);
    expect(controlMock.mock.calls[0][0]).toBe('playPause');
  });
});
