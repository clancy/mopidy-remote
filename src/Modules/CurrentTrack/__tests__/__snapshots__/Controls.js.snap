// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls tests renders disabled icons correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 25,
          },
          Object {
            "color": "white",
            "marginLeft": 10,
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 35,
          },
          Object {
            "color": "white",
            "marginLeft": 10,
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "white",
        "borderRadius": 35,
        "borderWidth": 0.5,
        "height": 70,
        "justifyContent": "center",
        "width": 70,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 50,
            },
            Object {
              "backgroundColor": "transparent",
              "color": "white",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 35,
          },
          Object {
            "color": "white",
            "marginLeft": 10,
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 25,
          },
          Object {
            "color": "white",
            "marginLeft": 10,
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Controls tests renders enabled icons correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 25,
          },
          Object {
            "color": "green",
            "marginLeft": 10,
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 35,
          },
          Object {
            "color": "white",
            "marginLeft": 10,
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "white",
        "borderRadius": 35,
        "borderWidth": 0.5,
        "height": 70,
        "justifyContent": "center",
        "width": 70,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 50,
            },
            Object {
              "backgroundColor": "transparent",
              "color": "white",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 35,
          },
          Object {
            "color": "white",
            "marginLeft": 10,
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 25,
          },
          Object {
            "color": "green",
            "marginLeft": 10,
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
